crates/turborepo-lib/src/client.rs

```diff
@@ -264,10 +264,13 @@ impl APIClient {
         false
     }
 
-    pub fn new(base_url: impl AsRef<str>) -> Result<Self> {
-        let client = reqwest::Client::builder()
-            .timeout(std::time::Duration::from_secs(20))
-            .build()?;
+    pub fn new(base_url: impl AsRef<str>, timeout: Option<u64>) -> Result<Self> {
+        let client = match timeout {
+            Some(timeout) => reqwest::Client::builder()
+                .timeout(std::time::Duration::from_secs(timeout))
+                .build()?,
+            None => reqwest::Client::builder().build()?,
+        };
```

Comment:
This is a material difference from an earlier iteration.

In Go if you pass "0" into things you get no timeout. In Rust to do that you have to avoid call `.timeout()`, so I match on an `Option` instead.